homeassistant/components/hyundai_kia_connect/__init__.py

```diff
@@ -0,0 +1,38 @@
+"""The Hyundai / Kia Connect integration."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import HyundaiKiaConnectDataUpdateCoordinator
+
+PLATFORMS: list[str] = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up Hyundai / Kia Connect from a config entry."""
+    """Set up this integration using UI."""
+    if hass.data.get(DOMAIN) is None:
+        hass.data.setdefault(DOMAIN, {})
+
+    coordinator = HyundaiKiaConnectDataUpdateCoordinator(hass, config_entry)
+    await coordinator.async_refresh()
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(
+        config_entry, PLATFORMS
+    )
+    if unload_ok:
+        hass.data[DOMAIN] = None
```

Comment:
This seems incorrect, if one set up 2 config entries/instances of the integration, it would kill both.

Reply:
fair callout, will fix in new revision